### PR TITLE
Rename data types to corresponding PostgreSQL data types.

### PR DESCRIPTION
**Pull request by kovrus**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java [hunk 1]
```diff
@@ -1606,6 +1607,12 @@ public Node visitDataType(SqlBaseParser.DataTypeContext context) {
             return CollectionColumnType.array((ColumnType) visit(context.arrayTypeDefinition().dataType()));
         } else if (context.setTypeDefinition() != null) {
             return CollectionColumnType.set((ColumnType) visit(context.setTypeDefinition().dataType()));
+        } else if (context.baseDataType() != null) {
```
Comment (kovrus) [inline sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java]: I'd probably change the parser a bit, e.g.:
```
ident
    : quotedIdentifier                                       #quotedIdentifierAlternative
    | unquotedIdentifier                                   #unquotedIdentifierAlternative
    ;

quotedIdentifier
    : QUOTED_IDENTIFIER                                  #quotedIdentifier
    | BACKQUOTED_IDENTIFIER                         #backQuotedIdentifier
    ;

unquotedIdentifier
    : IDENTIFIER                                               #unquotedIdentifier
    | DIGIT_IDENTIFIER                                    #digitIdentifier
    | nonReserved                                            #unquotedIdentifier
    | COLON_IDENT                                         #colonIdentifier
    ;

...
dataType
    : unquotedIdentifier
    | baseDataType
    | objectTypeDefinition
    | arrayTypeDefinition
    | setTypeDefinition
    ;

baseDataType
    : DOUBLE PRECISION
    ;
```

and implement visitors for correspoding nodes in `dataType` instead of having the `visitDataType`, but would do it in the separate PR.

File: blackbox/docs/general/ddl/data-types.rst [hunk 2]
```diff
@@ -104,39 +104,38 @@ Columns of type string can also be analyzed. See :ref:`sql_ddl_index_fulltext`.
...
+| ``smallint``         | 2 bytes  | small-range integer         | -32,768 to 32,767           |
++----------------------+----------+-----------------------------+-----------------------------+
+| ``integer``          | 4 bytes  | integer                     | -2^31 to 2^31-1.            |
++----------------------+----------+-----------------------------+-----------------------------+
+| ``bigint``           | 8 bytes  | large-range integer         | -2^63 to 2^63-1             |
++----------------------+----------+-----------------------------+-----------------------------+
+| ``real``             | 4 bytes  | inexact, variable-precision | 6 decimal digits precision  |
++----------------------+----------+-----------------------------+-----------------------------+
+| ``double precision`` | 8 bytes  | inexact, variable-precision | 15 decimal digits precision |
++----------------------+----------+-----------------------------+-----------------------------+
```
Comment (seut) [inline blackbox/docs/general/ddl/data-types.rst]: `char` is missing in that matrix or it links to the wrong section

File: blackbox/docs/general/ddl/data-types.rst [hunk 3]
```diff
@@ -717,6 +728,15 @@ brackets (``[]``), for example::
 
 .. _type_conversion:
 
+Special Character Types
```
Comment (seut) [inline blackbox/docs/general/ddl/data-types.rst]: the RST link `_type_conversion` above must be on the right title (shift down)

File: blackbox/docs/general/ddl/data-types.rst [hunk 4]
```diff
@@ -34,13 +34,13 @@ These are values that are atomic, not composed of separate parts, no containers
 or collections.
 
 * `boolean`_
-* `byte <numeric types_>`_
-* `short <numeric types_>`_
+* `char <numeric types_>`_
```
Comment (seut) [inline blackbox/docs/general/ddl/data-types.rst]: I think the link is not correct anymore?

File: sql-parser/src/main/antlr/SqlBase.g4 [hunk 5]
```diff
@@ -531,12 +531,17 @@ rerouteOption
...
     : IDENTIFIER
     | quotedIdentifier
+    | baseDataType
     | nonReserved
     | objectTypeDefinition
     | arrayTypeDefinition
     | setTypeDefinition
     ;
 
+baseDataType
```
Comment (seut) [inline sql-parser/src/main/antlr/SqlBase.g4]: why is it named `base...` ?

File: common/src/main/java/io/crate/types/DataTypes.java [hunk 6]
```diff
@@ -333,17 +340,30 @@ public static DataType ofName(String name) {
...
+        entry(LONG.id(), "long"),
+        entry(ObjectType.ID, "object"),
+        entry(GEO_SHAPE.id(), "geo_shape"),
+        entry(GEO_POINT.id(), "geo_point")
+    );
+
     @Nullable
-    public static DataType ofMappingName(String name) {
-        return MAPPING_NAMES_TO_TYPES.get(name);
+    public static String esMappingNameFrom(int typeId) {
```
Comment (seut) [inline common/src/main/java/io/crate/types/DataTypes.java]: what about adding a e.g. `mapperName()` method to `DataType` instead?

Comment (kovrus) [reply common/src/main/java/io/crate/types/DataTypes.java]: i thought about it, but it would also touch some unrelated `DataType` child classes like in aggregations, e.g. `AverageStateType`, collection types, etc. We can, of course, have have a default implementation if `mapperName` that returns probably null, but is it really worth to change the `DataType` interface in this case. It does not enforce an implementation of `mapperName`.

File: sql-parser/src/main/antlr/SqlBase.g4 (hunk 5 above)
Comment (kovrus) [reply sql-parser/src/main/antlr/SqlBase.g4]: definedDataType?

File: blackbox/docs/general/ddl/data-types.rst (hunk 2 above)
Comment (kovrus) [reply blackbox/docs/general/ddl/data-types.rst]: fixed the link, `char` goes to `special character types`

File: blackbox/docs/config/cluster.rst [hunk 7]
```diff
@@ -168,9 +168,9 @@ Collecting Stats
   Defines the refresh interval to refresh tables statistics used to produce
   optimal query execution plans.
 
-  This field expects a time value either as a long or double or alternatively
-  as a string literal with a time suffix (``ms``, ``s``, ``m``, ``h``, ``d``,
-  ``w``).
+  This field expects a time value eitheras a ``bigint`` or
```
Comment (smakalias) [inline blackbox/docs/config/cluster.rst]:
```suggestion
  This field expects a time value either as a ``bigint`` or
```
